content/courses/Ada_For_The_C_Embedded_Developer/chapters/06_Translation.rst

```diff
@@ -622,8 +622,8 @@ pieces of data into a larger structure. In Ada, this can be done by describing
 the structure layout at the type level through representation clauses, and then
 accessing this structure as any other.
 
-This is the case of an array of bytes representing various flags. In C, this
-would be done through masks, e.g.:
+Consider the case of using a C primitive type's bits to represent various
```

Comment:
I agree the original phrase isn't good. However, this part:

> using a C primitive type's bits

is a little bit confusing.

Reply:
https://en.wikipedia.org/wiki/Bit_field#C_programming_language
https://en.cppreference.com/w/cpp/language/bit_field

The type defines the size of the memory into which the bitfields will be mapped (and implementation specific of how those bitfields are interpreted as signed or unsigned).

Perhaps it could be reworded to something like:
" Consider the case of using a C primitive type as a container for single bit boolean flags"?

Reply:
Yes, the intention was actually clear. ;-) Just that specific expression ("a C primitive type's bits") was confusing. But I do like your rewording; you could use that.